lib/util/yaml.ts

```diff
@@ -32,8 +40,18 @@ export function parseYaml<ResT = unknown>(
 
   const parsed: ResT[] = [];
   for (const element of rawDocuments) {
-    const singleParsed = schema.parse(element);
-    parsed.push(singleParsed);
+    try {
+      const singleParsed = schema.parse(element);
```

Comment:
use safeParse, then we don't need the try-catch block

Reply:
If we catch the error we can use it to directly set the cause while re throwing.

Reply:
cause doesn't need to be an error object, can be anything 🤗

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause#description